Design.md

```diff
@@ -309,3 +332,33 @@ Get the platform usage of global (without query parameter), tenant (with tenant_
         * `key` : `offset` and `value` : `marker`
     1. `AssumeRoleCache` for all the `User` and `S3 Credential` APIs
         * `key` : `RoleArn` and `value` : `Credentials`
+
+## SecretKey Encryption Strategy
+
+1. Secret keys are encrypted and decrypted using a cipher algorithm and secret password provided in the `crypto.yml` config file.
+1. A `key-id` property will be provided for each `secret-password` in the `crypto.yml` config file to maintain the version of the key and will be used in the Key rotation.
+1. Meta-information will be encoded with base64 and added at the beginning of the encrypted string. This meta information will include the `key-id` of the `secret-password` used.
+1. If a `nonce` exists for the encrypted string, it will be added after the meta information in the result string.
```

Comment:
A nonce will always exist, and could be part of the meta-information. I wouldn't structure this as "Stuff is stored by encoding key-id using some mechanism, then append the nonce to it, then append the encrypted string to it". Instead, define a (versioned? Or protobuf-style?) structure which contains all the information required (key-id, nonce, encrypted string,...) and store a serialization of this.

Reply:
Fixed in 1fce348510aa482965c5a3a6ae1ea93beb31879d

Reply:
Modified serialization issues as suggested in 475cacd3a3e852b66ca20ae6f5320ae38faf75f9